modules/storages/app/common/storages/peripherals/storage_interaction/nextcloud/util.rb

```diff
@@ -71,7 +71,8 @@ def origin_user_id(caller:, storage:, auth_strategy:)
               when :basic_auth
                 ServiceResult.success(result: storage.username)
               when :oauth_user_token, :sso_user_token
-                origin_user_id = RemoteIdentity.where(user_id: auth_strategy.user, oauth_client: storage.oauth_client)
+                origin_user_id = RemoteIdentity.where(user_id: auth_strategy.user, auth_source: storage.oauth_client,
```

Comment:
:red_circle: This will also not fly for `sso_user_token` (which is part of this `when`), because authentication through SSO will have a different `auth_source`.

Reply:
Yes, that is right. I suggested above(in the uniqueness thread) to have two separate branches here with explicit auth_source set. Wdyt?